fsimpl/test/filetest.go

```diff
@@ -315,3 +315,32 @@ func testSameFile(t *testing.T, f1, f2 p9.File) {
 		t.Errorf("Attributes of same files do not match: %v and %v", f1Attr, f2Attr)
 	}
 }
+
+func TestReadWriteFS(t *testing.T, attach p9.Attacher) {
+	root, err := attach.Attach()
+	if err != nil {
+		t.Fatalf("Failed to attach to %v: %v", attach, err)
+	}
+	defer root.Close()
+
+	// Create a dummy file to unlink
+	dummyFileName := "dummy_file.txt"
+	_, _, _, err = root.Create(dummyFileName, p9.ReadWrite, 0666, p9.NoUID, p9.NoGID)
+	if err != nil {
+		t.Fatalf("Create failed: %v", err)
+	}
+	// Attempt to unlink the file
+	err = root.UnlinkAt(dummyFileName, 0)
+	if err != nil {
+		t.Errorf("Unlinkat failed: %v", err)
+	}
+	// Check that it does not exist anymore
+	dirents, err := readdir(root)
+	if err != nil {
+		t.Fatal(err)
+	}
+	file2dirent := dirents.Find("dummy_file.txt")
```

Comment:
This should use the name reference above instead of a literal string.
```suggestion
	file2dirent := dirents.Find(dummyFileName)
```